### DISCUSSION TITLE
Line Highlighting in html code block

### DISCUSSION THREAD

**Discussion by aborruso** [quarto-dev/quarto-cli#2987] [Q&A]
Hi,
for reveal output, I can use `{.python code-line-numbers="7,9"}` to highlight line 7 and 9.

If use it for a block code in an HTML output, I have no highlight for those lines.
Is there a way to do it also for HTML output?

Thank you

![image](https://user-images.githubusercontent.com/30607/197382095-16b41e27-cd85-405c-a565-c91758124eba.png)

**Comment by mcanouil** [comment]
I am not sure there is an easy way right now ...
The code line highlighting is a Reveal.js feature.
It could be ported to standard HTML format in theory, thus a feature request to Quarto team.

**Reply by aborruso** [reply]
What's the best way to create a feature request? Here in discussions or a new issue?

**Reply by mcanouil** [reply]
I do not know if there is official/preferred way, but currently there is two approaches:
- Wait for a Quarto team member to see this discussion, he will discuss it with the team, then if that's something they'll consider doing, they'll create an issue (feature request) from this discussion.
- Don't wait and open an issue with usually a tag in the title such as "[FR]" or "[Feature Request]" which describes as best as you can what you would like and possible hints/solutions to do that.

**Reply by shafayetShafee** [reply]
I have created a small Quarto filter extension [`line-highlight`](https://github.com/shafayetShafee/line-highlight) to enable line highlighting (both for source code and output) in HTML output format (i.e. `format: html`).

To highlight a specific line of source code, you need to use the chunk option `source-line-numbers` which takes values similarly as [`code-line-numbers`](https://quarto.org/docs/reference/formats/html.html#code). **Note that, this filter extension requires Quarto v1.2 at least to work.**

An example,

~~~
---
title: "Line Highlighting in HTML"
format: html
filters: 
  - line-highlight
jupyter: python3
---

```{python}
#| source-line-numbers: "3"
#| class-source: "numberLines"

import numpy as np

r = np.arange(0, 2, 0.01)
theta = 2 * np.pi * r
```
~~~

![line highlight](https://user-images.githubusercontent.com/48413872/211149116-7e6d9a5b-c23f-4b1e-be77-9fdcdc61f1a4.png)

See the [github repository](https://github.com/shafayetShafee/line-highlight#line-highlight-extension-for-quarto) for more examples and the installation process.

**Reply by aborruso** [reply]
Thank you @shafayetShafee
